pkg/cmd/extension/manager.go

```diff
@@ -194,10 +195,24 @@ func (m *Manager) populateLatestVersions(exts []*Extension) {
 func (m *Manager) InstallLocal(dir string) error {
 	name := filepath.Base(dir)
 	targetLink := filepath.Join(m.installDir(), name)
+	cs := m.io.ColorScheme()
+
 	if err := os.MkdirAll(filepath.Dir(targetLink), 0755); err != nil {
 		return err
 	}
-	return makeSymlink(dir, targetLink)
+	if err := makeSymlink(dir, targetLink); err != nil {
+		return err
+	}
+
+	// Check if an executable of the same name exists in the target directory.
+	// An error here doesn't indicate a failed extension installation, but
+	// it does indicate that the user will not be able to run the extension until
+	// the executable file is built or created manually somehow.
+	if _, err := os.Stat(filepath.Join(dir, name)); err != nil {
+		errMsg := fmt.Errorf("%v %w: executable file named \"%s\" in %s is required to run the extension after install. Perhaps you need to build it?", cs.WarningIcon(), ErrExtensionExecutableNotFound, name, dir)
```

Comment:
Maybe just return `fmt.Errorf`, since the variable is only used once, and it's not really a message, it's an error.

You could also avoid this whole wrapping business and do:

```go
type ErrExtensionExecutableNotFound struct {
 Dir string
 Name string
}
```

And either construct the error in `Error` method, or at the call site, if that's the only place it is used.

No particularly strong feelings here though, just offering an alternative idea.

Reply:
Thanks @williammartin 🙇 

I took a stab at your suggested alternative in b2ab7b84f84887e2757fe28be6ca616e2582f853 - I think it looks better in general